doc/library/config.txt

```diff
@@ -416,32 +416,36 @@ import theano and print the config variable, as in:
     `amdlibm <http://developer.amd.com/cpu/libraries/libm/>`__
     library, which is faster than the standard libm.
 
-.. attribute:: config.lib.cnmem
+.. attribute:: config.gpuarray.preallocate
 
-    Float value: >= 0
+    Float value
 
-    Controls the use of `CNMeM <https://github.com/NVIDIA/cnmem>`_ (a
-    faster CUDA memory allocator). In Theano dev version until 0.8
-    is released.
+    Default: 0
 
-    The CNMeM library is included in Theano and does not need to be
-    separately installed.
+    Controls the preallocation of memory with the gpuarray backend.
 
-    The value represents the start size (either in MB or the fraction of total GPU
-    memory) of the memory pool. If more memory is needed, Theano will
-    try to obtain more, but this can cause memory fragmentation.
+    The value represents the start size (either in MB or the fraction
+    of total GPU memory) of the memory pool. If more memory is needed,
+    Theano will try to obtain more, but this can cause memory
+    fragmentation.
 
-        * 0: not enabled.
-        * 0 < N <= 1: use this fraction of the total GPU memory (clipped to .95 for driver memory).
+    A negative value will completely disable the allocation cache.
+    This can have a severe impact on performance and so should not be
+    done outside of debugging.
+
+        * < 0: disabled
+        * 0 <= N <= 1: use this fraction of the total GPU memory (clipped to .95 for driver memory).
         * > 1: use this number in megabytes (MB) of memory.
```

Comment:
In both back-end, we clip at 0.95 if it is between 0 and 1. This is needed to let memory for the driver and other lib that preallocate memory themself.